board/safety/safety_chrysler.h

```diff
@@ -0,0 +1,129 @@
+// board enforces
+//   in-state
+//      ACC is active (green)
+//   out-state
+//      brake pressed
+//      stock LKAS ECU is online
+//      ACC is not active (white or disabled)
+
+// chrysler_: namespacing
+int chrysler_speed = 0;
+
+// silence everything if stock ECUs are still online
+int chrysler_lkas_detected = 0;
+
+int chrysler_ignition_started = 0;
+
+static void chrysler_rx_hook(CAN_FIFOMailBox_TypeDef *to_push) {
+  int bus_number = (to_push->RDTR >> 4) & 0xFF;
+  uint32_t addr;
+  if (to_push->RIR & 4) {
+    // Extended
+    // Not looked at, but have to be separated
+    // to avoid address collision
+    addr = to_push->RIR >> 3;
+  } else {
+    // Normal
+    addr = to_push->RIR >> 21;
+  }
+
+  // Gear selector used for determining ignition.
```

Comment:
instead of gear I would use any CAN message that represents "car on"

Reply:
I actually switched it to use the default_ign_hook. The chrysler giraffe only has power if the car's ignition is on, and I've wired that 12V to the IGN GPIO on panda.